### PR TITLE
Switched to file based namespaces and targeted type new expressions (Chapter 7)

### PR DESCRIPTION
**Pull request by MarkMichaelis**

### CODE REVIEW

File: src/Chapter07/Listing07.07.ProtectedMembersAreAccessibleOnlyFromDerivedClasses.cs [hunk 1]
```diff
@@ -1,59 +1,58 @@
...
+        // for the filename
         #region HIGHLIGHT
-        protected Guid ObjectKey { get; }
+        using FileStream stream = File.OpenWrite(
+            ObjectKey + ".dat");
         #endregion HIGHLIGHT
+        // ...
+        stream.Dispose();
     }
+    static public Contact Copy(Contact contact)
```
Comment (Keboo) [inline src/Chapter07/Listing07.07.ProtectedMembersAreAccessibleOnlyFromDerivedClasses.cs:34]:
```suggestion
    public static Contact Copy(Contact contact)
```

File: src/Chapter07/Listing07.20.TuplePatternMatching.cs [hunk 2]
```diff
@@ -1,42 +1,41 @@
...
+namespace AddisonWesley.Michaelis.EssentialCSharp.Chapter07.Listing07_20;
+
+using System.IO;
+
+#region INCLUDE
+public class Program
 {
-    using System.IO;
+    const int Action = 0;
+    const int FileName = 1;
```
Comment (Keboo) [inline src/Chapter07/Listing07.20.TuplePatternMatching.cs:11]: Should these have access modifiers specified?

File: src/Chapter07/Listing07.25.DataConversionUsingTheAsOperator.cs [hunk 3]
```diff
@@ -1,39 +1,38 @@
...
-        protected Guid ObjectKey { get; }
-        // ...
-    }
+public class Contact : PdaItem
+{
+    // ...
+    public Contact(string name) => Name = name;
 
-    public class Contact : PdaItem
+    static public Contact Load(PdaItem pdaItem)
```
Comment (Keboo) [inline src/Chapter07/Listing07.25.DataConversionUsingTheAsOperator.cs:17]:
```suggestion
    public static Contact Load(PdaItem pdaItem)
```

**Review by Keboo — APPROVED** [review]

**Review by Keboo — APPROVED** [review]

File: src/Chapter07/Listing07.20.TuplePatternMatching.cs (hunk 2 above)
Comment (MarkMichaelis) [reply src/Chapter07/Listing07.20.TuplePatternMatching.cs:11]:
```suggestion
    private const int Action = 0;
    private const int FileName = 1;
```

**Pull request merged**